### ISSUE TITLE
Memory leak within download_file()

### ISSUE THREAD

**Issue by Nemris** [llakssz/FunKiiU#40]
Good day. While using FunKiiU on Android to download moderately sized titles, the device's RAM was depleted, and the majority of background tasks killed. In turn, the RAM was freed when the script terminated its execution, either normally or otherwise.

By experimenting a bit, I found a memory leak in the function:

    download_file(url, outfname, retry_count=3, ignore_404=False, expected_size=None, chunk_size=0x4096)

Basically, you open a socket `infile` when preparing to download a file, but never close it upon completion or failure; this results in a severe memory hogging during the whole FunKiiU execution. It can be fixed by `infile.close()` -ing the socket when you're done with it.

**Comment by llakssz** [comment]
Interesting, thank you for noticing this and even saying how to fix it.
I'll work on it soon.

**Comment by Nemris** [comment]
Excellent. Since I now have your attention, I would like to inform you that I'm rewriting FunKiiU from scratch, under a different name and license.

It will include slightly different features, such as the ability to resume the download of a partially downloaded file, and it will probably skim down on different things, such as Python 2 support and bulk ticket download per region.

Is this acceptable?

**Comment by llakssz** [comment]
By all means do what you want! That sounds very cool, I have been busy recently and not had too much time for projects until now.
Good luck with your program 🆒

**Comment by Nemris** [comment]
Perfect. Good luck with your projects!

**Comment by V10lator** [comment]
Should be fixed at https://pastebin.com/5tNswBAX
Changelog:
- Remove Python 2 support
- Resume downloads
- Handle server / connection errors correctly
- Allow numbers higher than 9 for the retry count
- Overwrite lines on CLI correctly
- Close file handlers correctly
- Return one in case of error so the CLI gets informed correctly
